### PR TITLE
update hierarchical BNN notebook with some nice figures

### PR DESCRIPTION
**Pull request by murphyk**
update hierarchical BNN notebook with some nice figures.
Code has not changed.

### CODE REVIEW

**Comment by rlouf** [comment]
Thanks @canyon289 ! We agreed over email that I'd do the merge myself. Should have commented here.

**Comment by canyon289** [comment]
Ah sorry. Carry on you two :) I'll get out of the way

File: notebooks/bnn_hierarchical_blackjax.ipynb [hunk 1]
```diff
@@ -35,7 +35,7 @@
       "source": [
         "# Hierarchical Bayesian neural networks\n",
         "\n",
```
Comment (rlouf) [inline notebooks/bnn_hierarchical_blackjax.ipynb:37]: This commit is not doing anything, and I don't see any difference with the notebook that is already on master. Are you sure you opened a PR with the right branch?

**Review by rlouf — CHANGES_REQUESTED** [review]
I can't find any difference between this notebook and the one on the `main` branch, maybe you didn't push the correct branch?

File: notebooks/bnn_hierarchical_blackjax.ipynb (hunk 1 above)
Comment (murphyk) [reply notebooks/bnn_hierarchical_blackjax.ipynb]: You are right, it was the wrong version. I've updated it.

**Comment by murphyk** [comment]
I fixed this and added the new notebook, but get this commit error

Run ***@***.***
nstall pre-commit
Cache Size: ~81 MB (85222484 B)
/usr/bin/tar --use-compress-program zstd -d -xf
/home/runner/work/_temp/9370fc14-9e34-4f63-bfeb-61c1a69d6231/cache.tzst -P
-C /home/runner/work/blackjax/blackjax
/opt/hostedtoolcache/Python/3.9.7/x64/bin/pre-commit run
--show-diff-on-failure --color=always --all-files
Check for merge
conflicts................................................Passed
Check Toml...........................................(no files to
check)Skipped
Check
Yaml...............................................................Passed
Debug Statements
(Python)................................................Passed
Fix End of
Files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook

Fixing notebooks/HierarchicalBNN.ipynb

Don't commit to
branch..................................................Skipped
Fix
requirements.txt.....................................................Passed
Trim Trailing
Whitespace.................................................Passed
isort....................................................................Passed
pyupgrade................................................................Passed
flake8...................................................................Passed
black....................................................................Passed
mypy.....................................................................Passed
nbqa-black...............................................................Failed
- hook id: nbqa-black
- files were modified by this hook

reformatted notebooks/HierarchicalBNN.ipynb
All done! ✨ 🍰 ✨
1 file reformatted, 5 files left unchanged.

nbqa-pyupgrade...........................................................Failed
- hook id: nbqa-pyupgrade
- exit code: 1
- files were modified by this hook

Rewriting notebooks/HierarchicalBNN.ipynb

nbqa-isort...............................................................Failed
- hook id: nbqa-isort
- files were modified by this hook

Fixing /home/runner/work/blackjax/blackjax/notebooks/HierarchicalBNN.ipynb

nbqa-flake8..............................................................Failed
- hook id: nbqa-flake8
- exit code: 1

notebooks/HierarchicalBNN.ipynb:cell_2:11:1: F401 'sklearn' imported but
unused
notebooks/HierarchicalBNN.ipynb:cell_2:13:1: F401 'jax.jit' imported but
unused
notebooks/HierarchicalBNN.ipynb:cell_2:14:1: F401 'jax.random.normal'
imported but unused
notebooks/HierarchicalBNN.ipynb:cell_2:15:1: F401 'sklearn.datasets'
imported but unused
notebooks/HierarchicalBNN.ipynb:cell_2:17:1: F401
'sklearn.model_selection.train_test_split' imported but unused
notebooks/HierarchicalBNN.ipynb:cell_22:9:9: F841 local variable 'contour'
is assigned to but never used
notebooks/HierarchicalBNN.ipynb:cell_26:3:5: F841 local variable
'output_dim' is assigned to but never used
notebooks/HierarchicalBNN.ipynb:cell_31:9:9: F841 local variable 'contour'
is assigned to but never used

pre-commit hook(s) made changes.
If you are seeing this message in CI, reproduce locally with: `pre-commit
run --all-files`.
To run `pre-commit` as part of git workflow, use `pre-commit install`.
All changes made by hooks:
diff --git a/notebooks/HierarchicalBNN.ipynb
b/notebooks/HierarchicalBNN.ipynb
index 9924d91..bd91796 100644
--- a/notebooks/HierarchicalBNN.ipynb
+++ b/notebooks/HierarchicalBNN.ipynb
@@ -1,1001 +1,1068 @@
 {

**Comment by rlouf** [comment]
`pre-commit` runs a notebook linter which found issue with the code. I fixed these when I merged your previous PR, I'll do the same here.

Versioning notebooks is a bit of a pain. I'll see if we can keep example notebooks as markdown files in the repo using `jupytext` and render them with figures in the documentation.

**Comment by rlouf** [comment]
I am merging and will deal with the errors in a new branch. Thank you so much for adding these figures!

**Pull request merged**